app/middleware/account_elevator.rb

```diff
@@ -11,7 +11,8 @@ def parse_tenant_name(request)
   end
 
   def self.switch!(cname)
-    account = Account.find_by(cname: Account.canonical_cname(cname))
+    account = Account.joins(:domain_names).find_by(domain_names: { is_active: true,
```

Comment:
you can use the `from_cname` class method here

Reply:
@smcfarlane good point - updated